python/cudf/cudf/core/_internals/expressions.py

```diff
@@ -120,7 +120,7 @@ def visit_Name(self, node):
         self.stack.append(ColumnReference(col_id))
 
     def visit_Constant(self, node):
-        if not isinstance(node, (ast.Num, ast.Str)):
+        if not isinstance(node.value, (float, int, str, complex)):
```

Comment:
The use of `ast.Num` here resulted in a deprecation warning that was treated as an error in tests (thanks @mroeschke !).

> DeprecationWarning: ast.Num is deprecated and will be removed in Python 3.14; use ast.Constant instead

([build link](https://github.com/rapidsai/cudf/actions/runs/10711254837/job/29737831951?pr=16745#step:8:3838))

I looked into this... seems `ast.Num` and other specialized constants have been deprecated since Python 3.8 and are set to be removed in Python 3.14.

This proposes checking the value of the `ast.Constant` being passed in here against a hard-coded set of builtins.

* more details on that deprecation: https://github.com/python/cpython/blob/aa1339aaaa6363c38186defaa079d069b4cb08b2/Doc/whatsnew/3.8.rst?plain=1#L1668-L1671
* evidence that this list of builtins is identical to what `ast.Num` and `ast.Str` were doing: https://github.com/python/cpython/blob/a50a86db0fd66c7ea53e56e346057be1153d6d47/Lib/ast.py#L528-L534

Reply:
This looks correct to me. Thanks.